src/sgame/sg_bot_util.cpp

```diff
@@ -613,8 +613,11 @@ AINodeStatus_t BotActionEvolve( gentity_t *self, AIGenericNode_t* )
 	{
 		evolveInfo_t info = BG_ClassEvolveInfoFromTo( currentClass, cl.item );
 
-		if ( info.evolveCost > 0 // no devolving or evolving to the same
-				&& BotEvolveToClass( self, cl.item ) )
+		bool isBuilder = currentClass == PCL_ALIEN_BUILDER0 || currentClass == PCL_ALIEN_BUILDER0_UPG;
+		if ( ( info.evolveCost > 0     // no devolving or evolving to the same
+			   || ( isBuilder && cl.item == PCL_ALIEN_LEVEL0 )
+			   || ( currentClass == PCL_ALIEN_BUILDER0 && cl.item == PCL_ALIEN_BUILDER0_UPG ) )
+			 && BotEvolveToClass( self, cl.item ) )
```

Comment:
Thanks for fixing this! I believe this is equivalent to this:

```suggestion
		if ( ( !info.isDevolving && BotEvolveToClass( self, cl.item ) )
```

Interestingly I had the same problem when indicating the evolutions in the UI.

Reply:
LGTM to this version

Reply:
I find this to be a little silly. First, we had evolve. Then we made an exception, and now we have devolve. Once we make the next exception, we will already be out of words.

Reply:
The distinction between evolving and devolving is that you have to be close to the overmind to devolve. 

Maybe we should allow any evolution or devolution though? Looking at `build.bt`, it seems to expect being able to devolve: 
```
                                                action moveTo( E_A_OVERMIND )
                                                selector
                                                {
                                                        action evolveTo( PCL_ALIEN_BUILDER0_UPG )
                                                        action evolveTo( PCL_ALIEN_BUILDER0 )
                                                }
```

Reply:
`evolveTo` can indeed devolve

Reply:
To clarify: this PR changes the BT action `evolve`, which is usually done by each alien bot at each frame. It lets the bot choose the class to evolve to. This is different from the action `evolveTo( CLASS )`, tells the bot to evolve to `CLASS`.

My dislike of the term "devolving" stems from the fact that "evolve" and "devolve" sound like opposites, when in fact devolve is simply an evolve with an additional condition to be satisfied. Consequently, most of our evolve code handles devolves gracefully. I would rather NUKE all mentions of "devolve", and simply embrace the fact that there are (dynamic) conditions to be satisfied for an evolve to succeed.

Reply:
Here is a (probably incomplete) list of reasons why your attempt to evolve/devolve can fail:

- You are too poor
- You are trying to get bigger, but the map geometry is blocking you
- You are trying to get bigger, but a player is blocking you
- You are trying to get bigger, but a building is blocking you
- Your wall climb is active
- You are currently attacking
- You are trying to devolve too far away from the overmind

In the end, we should not make bots try to handle all these cases. It is best to make them run around and try until they succeed.